File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.1/Java/src/Main.java
public class Main {
  public static int sum(int[] list, int start) {
    if (start >= list.length) {
      return 0;
    } else {
      return list[start] + sum(list, start + 1);
    }
  }

  public static void main(String[] args) {
    int[] list = {1, 2, 3, 4};
    int result = sum(list, 0);
    System.out.println(result);

    list = new int[]{2, 5, 6, 7};
    result = sum(list, 0);
    System.out.println(result);

    list = new int[]{3, 5, 7, 9};
    result = sum(list, 0);
    System.out.println(result);
  }
}

File: Capitulo-04/Dividir-Para-Conquistar/Exercicios/4.1/Rust/src/main.rs
fn sum(lista: &[i32]) -> i32 {
    if lista.is_empty() {
        0
    } else {
        lista[0] + sum(&lista[1..])
    }
}

fn main() {
    let mut result = sum(&[1, 2, 3, 4]);
    println!("{}", result);

    result = sum(&[2, 5, 6, 7]);
    println!("{}", result);

    result = sum(&[3, 5, 7, 9]);
    println!("{}", result);
}
